test/inductor/test_torchinductor_dynamic_shapes.py

```diff
@@ -609,8 +610,92 @@ def func(x, fn, a):
         output = cfunc(x, op, a)
         self.assertEqual(output, expected)
 
+    def test_wrapper_codegen_statically_known_int_or_none(self):
+        torch._dynamo.reset()
+
+        _x = torch.randn([5, 3, 3])
+        torch._dynamo.mark_dynamic(_x, 0)
+
+        # Simple functions introducing constraints on x.shape[0]
+        def fn_1(x):
+            # no constraint
+            return x.sin()
+
+        def fn_2(x):
+            # constrain in two directions
+            if x.shape[0] > 5:
+                return x.cos()
+            if x.shape[0] < 5:
+                return x * 2
+            # x.shape[0] == 5 at this point
```

Comment:
Funnily enough, this should also error out when using `mark_dynamic`. Mind filing an issue? cc @ezyang

Reply:
cc @avikchaudhuri

Reply:
Filed https://github.com/pytorch/pytorch/issues/122307